src/storage-controller/src/lib.rs

```diff
@@ -1726,6 +1760,15 @@ where
         // IDs of sources (and subsources) whose statistics should be cleared.
         let mut source_statistics_to_drop = vec![];
 
+        while let Some(Ok(dropped_ids)) = self
```

Comment:
With my suggestion above, this would just become a `try_recv()` on the channel of ready table drops.

Reply:
Done